pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -800,7 +800,9 @@ public void checkEncryption() {
 
     public void publishTxnMessage(TxnID txnID, long producerId, long sequenceId, long highSequenceId,
                                   ByteBuf headersAndPayload, long batchSize, boolean isChunked, boolean isMarker) {
-        checkAndStartPublish(producerId, sequenceId, headersAndPayload, batchSize, null);
+        if (!checkAndStartPublish(producerId, sequenceId, headersAndPayload, batchSize, null)) {
+            return;
+        }
```

Comment:
Is this its own bug? It seems unrelated to encryption.

Reply:
yes it's a bit out of the encryption fix scope. I can open another pull request

Reply:
removed. will open another pull